src/Client/Resources/keen-io-3_0.php

```diff
@@ -1247,6 +1247,146 @@
+                ),
+                'key'      => array(
+                    'location'    => 'uri',
+                    'type'        => 'string',
+                    'required'    => true,
+                ),
+                'masterKey' => array(
+                    'location'    => 'header',
+                    'description' => 'The Master API Key.',
+                    'sentAs'      => 'Authorization',
+                    'pattern'     => '/^([[:alnum:]])+$/',
+                    'type'        => 'string',
+                    'required'    => true,
+                ),
+            ),
+        ),
+        'updateAccessKey' => array(
+            'uri'         => 'projects/{projectId}/keys/{key}',
+            'description' => 'Updates a project access key.',
+            'httpMethod'  => 'POST',
+            'parameters'  => array(
+                'projectId'        => array(
+                    'location'    => 'uri',
+                    'type'        => 'string'
+                ),
+                'key'      => array(
+                    'location'    => 'uri',
+                    'type'        => 'string',
+                    'required'    => true,
+                ),
+                'masterKey' => array(
+                    'location'    => 'header',
+                    'description' => 'The Master API Key.',
+                    'sentAs'      => 'Authorization',
+                    'pattern'     => '/^([[:alnum:]])+$/',
+                    'type'        => 'string',
+                    'required'    => true,
+                ),
+            ),
+            'additionalParameters' => array(
+                'location' => 'json'
+            ),
+        ),
+        'revokeAccessKey' => array(
+            'uri'         => 'projects/{projectId}/keys/{key}/revoke',
+            'description' => 'Revokes a project access key.',
+            'httpMethod'  => 'POST',
+            'parameters'  => array(
+                'projectId'        => array(
+                    'location'    => 'uri',
+                    'type'        => 'string'
+                ),
+                'key'      => array(
+                    'location'    => 'uri',
+                    'type'        => 'string',
+                    'required'    => true,
+                ),
+                'masterKey' => array(
+                    'location'    => 'header',
+                    'description' => 'The Master API Key.',
+                    'sentAs'      => 'Authorization',
+                    'pattern'     => '/^([[:alnum:]])+$/',
+                    'type'        => 'string',
+                    'required'    => true,
+                ),
+            ),
+        ),
+        'unRevokeAccessKey' => array(
+            'uri'         => 'projects/{projectId}/keys/{key}/unrevoke',
+            'description' => 'Unrevokes a project access key.',
+            'httpMethod'  => 'POST',
+            'parameters'  => array(
```

Comment:
Would be cool if these parameters weren't duplicated but defined elsewhere. Seems like there's a common need for having the `projectId` and `masterKey` for all access key operations, and operations dealing with a single access key all need `key` in addition to those, so there might be just two variations of those parameters.

Reply:
@baumatron I am refactoring completely... reused params are moved to a separate class, also I added many constants instead of keys, this should make easier to use params and to avoid syntax errors in key typing